internal/server/authz/engine/bundle/engine.go

```diff
@@ -0,0 +1,98 @@
+package bundle
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/open-policy-agent/contrib/logging/plugins/ozap"
+	"github.com/open-policy-agent/opa/sdk"
+	"github.com/open-policy-agent/opa/storage/inmem"
+	"go.flipt.io/flipt/internal/config"
+	"go.flipt.io/flipt/internal/server/authz"
+	"go.uber.org/zap"
+)
+
+var _ authz.Verifier = (*Engine)(nil)
+
+type cleanupFunc func()
+
+type Engine struct {
+	opa          *sdk.OPA
+	logger       *zap.Logger
+	cleanupFuncs []cleanupFunc
+}
+
+func NewEngine(ctx context.Context, logger *zap.Logger, cfg *config.Config) (*Engine, error) {
+	var (
+		opaConfig    string
+		cleanupFuncs []cleanupFunc
+	)
+
+	switch cfg.Authorization.Backend {
+	case config.AuthorizationBackendObject:
+		opaConfig = cfg.Authorization.Object.String()
+
+		switch cfg.Authorization.Object.Type { //nolint
+		case config.S3ObjectAuthorizationBackendType:
+			// set AWS_REGION env var if not set and region is specified
+			// this is a nicety as the OPA env credentials provider requires this env var
+			// to be set, but we don't want the user to have to supply it twice if they already have it in the config
+			if cfg.Authorization.Object.S3.Region != "" && os.Getenv("AWS_REGION") == "" {
+				os.Setenv("AWS_REGION", cfg.Authorization.Object.S3.Region)
+				cleanupFuncs = append(cleanupFuncs, func() {
+					os.Unsetenv("AWS_REGION")
+				})
+			}
+		}
+	case config.AuthorizationBackendBundle:
+		opaConfig = cfg.Authorization.Bundle.String()
+	}
+
+	level, err := zap.ParseAtomicLevel(cfg.Log.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	opa, err := sdk.New(ctx, sdk.Options{
+		Config: strings.NewReader(opaConfig),
+		Store:  inmem.New(),
+		Logger: ozap.Wrap(logger, &level),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Engine{
+		logger:       logger,
+		opa:          opa,
+		cleanupFuncs: cleanupFuncs,
+	}, nil
+}
+
+func (e *Engine) IsAllowed(ctx context.Context, input map[string]interface{}) (bool, error) {
+	e.logger.Debug("evaluating policy", zap.Any("input", input))
+	dec, err := e.opa.Decision(ctx, sdk.DecisionOptions{
+		Path:  "flipt/authz/v1/allow",
+		Input: input,
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	allow, ok := dec.Result.(bool)
+	if !ok || !allow {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
Take it or leave it:
```suggestion
	allow, _ := dec.Result.(bool)
	return allow, nil
```